pixel/image.go

```diff
@@ -105,14 +139,15 @@ func (img Image[T]) setPixel(index int, c T) {
 	case zeroColor.BitsPerPixel() == 1:
 		// Monochrome.
 		x := index % int(img.width)
-		y := index / int(img.width)
-		offset := x + (y/8)*int(img.width)
+		offset := index / 8
```

Comment:
You are also changing the format here! The offset calculation changed and now outputs different values.

Reply:
See above comment.